src/components/button-toggle/button-toggle-group/button-toggle-group.test.tsx

```diff
@@ -0,0 +1,424 @@
+        <div>Foo</div>
+      </ButtonToggleGroup>
+    )
+  ).toThrow("`ButtonToggleGroup` only accepts children of type `ButtonToggle`");
+});
+
+test("should render with provided label", () => {
+  render(
+    <ButtonToggleGroup
+      id="button-toggle-group-id"
+      label="Group Label"
+      onChange={() => {}}
+    >
+      <ButtonToggle value="foo">Foo</ButtonToggle>
+      <ButtonToggle value="bar">Bar</ButtonToggle>
+    </ButtonToggleGroup>
+  );
+
+  expect(screen.getByText("Group Label")).toBeVisible();
+});
+
+test("should render with provided hintText", () => {
+  render(
+    <ButtonToggleGroup
+      id="button-toggle-group-id"
+      inputHint="Group Hint Text"
+      onChange={() => {}}
+    >
+      <ButtonToggle value="foo">Foo</ButtonToggle>
+      <ButtonToggle value="bar">Bar</ButtonToggle>
+    </ButtonToggleGroup>
+  );
+
+  expect(screen.getByText("Group Hint Text")).toBeVisible();
+});
+
+test("should render group with aria-label and no aria-labelledby when the label is not set", () => {
+  render(
+    <ButtonToggleGroup
+      id="button-toggle-group-id"
+      aria-label="Group Aria Label"
+      onChange={() => {}}
+    >
+      <ButtonToggle value="foo">Foo</ButtonToggle>
+      <ButtonToggle value="bar">Bar</ButtonToggle>
+    </ButtonToggleGroup>
+  );
+
+  expect(screen.getByRole("group")).toHaveAttribute(
+    "aria-label",
+    "Group Aria Label"
+  );
+  expect(screen.getByRole("group")).not.toHaveAttribute("aria-labelledby");
+});
+
+test("should render group with aria-labelledby set to the label's id and no aria-label when the label is set", () => {
+  render(
+    <ButtonToggleGroup
+      id="button-toggle-group-id"
+      label="Group Label"
+      aria-label="Group Aria Label"
+      onChange={() => {}}
+    >
+      <ButtonToggle value="foo">Foo</ButtonToggle>
+      <ButtonToggle value="bar">Bar</ButtonToggle>
+    </ButtonToggleGroup>
+  );
+
+  expect(screen.getByText("Group Label")).toHaveAttribute("id", "guid-12345");
+  expect(screen.getByRole("group")).toHaveAttribute(
+    "aria-labelledby",
+    "guid-12345"
+  );
+  expect(screen.getByRole("group")).not.toHaveAttribute("aria-label");
+});
+
+test("should render ButtonToggle children with aria-describedby set to the inputHint's id when the inputHint is set", () => {
+  render(
+    <ButtonToggleGroup
+      id="button-toggle-group-id"
+      inputHint="Group Hint Text"
+      onChange={() => {}}
+    >
+      <ButtonToggle value="foo">Foo</ButtonToggle>
+      <ButtonToggle value="bar">Bar</ButtonToggle>
+    </ButtonToggleGroup>
+  );
+
+  expect(screen.getByText("Group Hint Text")).toHaveAttribute(
+    "id",
+    "guid-12345"
+  );
+  expect(screen.getByRole("button", { name: "Foo" })).toHaveAttribute(
+    "aria-describedby",
+    "guid-12345"
+  );
+  expect(screen.getByRole("button", { name: "Bar" })).toHaveAttribute(
+    "aria-describedby",
+    "guid-12345"
+  );
+});
```

Comment:
Great job using the `*ByRole` queries to get the group element :+1:

**suggestion**: since jest-dom providers the matchers `toHaveAccessibleName` and `toHaveAccessibleDescription` for asserting the name and description of an element, we can simplify the tests further by not needing to mock the `guid()` helper. This will make the intended behaviour clearer:

```tsx
test("uses aria-label prop as its accessible name when passed", () => {
  render(
    <ButtonToggleGroup
      id="button-toggle-group-id"
      aria-label="Group Aria Label"
      onChange={() => {}}
    >
      <ButtonToggle value="foo">Foo</ButtonToggle>
      <ButtonToggle value="bar">Bar</ButtonToggle>
    </ButtonToggleGroup>
  );

  expect(screen.getByRole("group")).toHaveAccessibleName("Group Aria Label");
});

test("uses label prop as its accessible name when passed", () => {
  render(
    <ButtonToggleGroup
      id="button-toggle-group-id"
      label="Group Label"
      aria-label="Group Aria Label"
      onChange={() => {}}
    >
      <ButtonToggle value="foo">Foo</ButtonToggle>
      <ButtonToggle value="bar">Bar</ButtonToggle>
    </ButtonToggleGroup>
  );

  expect(screen.getByRole("group")).toHaveAccessibleName("Group Label");
});

test("uses inputHint prop as the accessible description for each child button when passed", () => {
  render(
    <ButtonToggleGroup
      id="button-toggle-group-id"
      inputHint="Group Hint Text"
      onChange={() => {}}
    >
      <ButtonToggle value="foo">Foo</ButtonToggle>
      <ButtonToggle value="bar">Bar</ButtonToggle>
    </ButtonToggleGroup>
  );

  expect(
    screen.getByRole("button", { name: "Foo" })
  ).toHaveAccessibleDescription("Group Hint Text");
  expect(
    screen.getByRole("button", { name: "Bar" })
  ).toHaveAccessibleDescription("Group Hint Text");
});
```